cli/tests/integrations/test_task.py

```diff
@@ -262,28 +259,59 @@ def test_log_master_unavailable():
 
 def test_log_slave_unavailable():
     """ Test slave's state.json being unavailable """
-    with app(SLEEP1, 'test-app', True):
-        client = mesos.DCOSClient()
-        client.get_slave_state = _mock_exception()
+    client = mesos.DCOSClient()
+    client.get_slave_state = _mock_exception()
 
-        with patch('dcos.mesos.DCOSClient', return_value=client):
-            args = ['task', 'log', 'test-app']
-            stderr = (b"""Error accessing slave: exception\n"""
-                      b"""No matching tasks. Exiting.\n""")
-            assert_mock(main, args, returncode=1, stderr=stderr)
+    with patch('dcos.mesos.DCOSClient', return_value=client):
+        args = ['task', 'log', 'test-app1']
+        stderr = (b"""Error accessing slave: exception\n"""
+                  b"""No matching tasks. Exiting.\n""")
+        assert_mock(main, args, returncode=1, stderr=stderr)
 
 
 def test_log_file_unavailable():
     """ Test a file's read.json being unavailable """
-    with app(SLEEP1, 'test-app', True):
-        files = _mesos_files(False, "", "stdout")
-        assert len(files) == 1
-        files[0].read = _mock_exception('exception')
+    fltr = "test-app1"
+    files = _mesos_files(False, fltr, "stdout")
+    assert len(files) == 1
+    files[0].read = _mock_exception('exception')
+
+    with patch('dcoscli.task.main._mesos_files', return_value=files):
+        args = ['task', 'log', fltr]
+        stderr = b"No files exist. Exiting.\n"
+        assert_mock(main, args, returncode=1, stderr=stderr)
+
+
+def test_ls():
+    assert_command(['dcos', 'task', 'ls', 'test-app1'],
+                   stdout=b'stderr  stdout\n')
+
+
+def test_ls_multiple_tasks():
+    stderr = b'Multiple tasks matching ID [test-app]: [test-app], ' + \
+             b'[test-app2]'
+    returncode, stdout, stderr = exec_command(
+        ['dcos', 'task', 'ls', 'test-app'])
```

Comment:
Hmm. Why does this pass? Should this be `b'Multiple tasks matching ID [test-app]: [test-app1], [test-app2]'`. Notice `test-app1`.